lib/plugins.ts

```diff
@@ -436,11 +428,21 @@ export class Plugins {
    * @return {webdriver.promise.Promise|Q.Promise} A promise which resolves to the
    *     function's return value
    */
-  safeCallPluginFun(
-      pluginObj: ProtractorPlugin, funName: string, args: IArguments, promiseType: PromiseType,
-      failReturnVal: any): any {
-    var deferred = promiseType == PromiseType.Q ? Q.defer() : webdriver.promise.defer();
-    var logError = (e: any) => {
+  private safeCallPluginFun(
+      pluginObj: ProtractorPlugin, funName: string, args: any[], promiseType: PromiseType,
+      failReturnVal: any) {
+    let resultPromise: webdriver.promise.Promise<any>|q.Promise<any>;
```

Comment:
The `'q'` issue should be fixed when your PR to DefinitelyTyped is merged. Is `fulfill` deprecated (https://github.com/kriskowal/q/blob/v1/q.js#L1141)? I believe that comment should be fixed in your DefinitelyTyped PR (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12358).